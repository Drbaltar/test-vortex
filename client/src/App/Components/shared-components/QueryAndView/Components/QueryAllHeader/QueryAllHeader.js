import React from 'react';
import QueryButton from './Components/QueryButton';

const QueryAllHeader = (props) => {
    const { hasQueryRan, query } = props;

    return <div className="pb-4">
        <QueryButton hasQueryRan={hasQueryRan} query={query}/>
    </div>;
};

export default QueryAllHeader;