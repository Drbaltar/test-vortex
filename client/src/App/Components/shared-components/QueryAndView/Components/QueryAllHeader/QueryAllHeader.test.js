import React from 'react';
import { shallow } from 'enzyme';
import QueryAllHeader from './QueryAllHeader';

const mockQueryFunction = jest.fn();

describe('QueryAllHeader', () => {
    const wrapper = shallow(<QueryAllHeader hasQueryRan={true} query={mockQueryFunction}/>);

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('passes boolean value indicating if query has previously been run', () => {
        expect(wrapper.find('QueryButton').prop('hasQueryRan')).toBe(true);
    });

    it('passes the query function to the QueryButton', () => {
        expect(wrapper.find('QueryButton').prop('query')).toEqual(mockQueryFunction);
    });
});