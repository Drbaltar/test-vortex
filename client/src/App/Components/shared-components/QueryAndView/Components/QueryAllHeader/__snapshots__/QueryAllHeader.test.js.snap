// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryAllHeader renders correctly 1`] = `
<div
  className="pb-4"
>
  <QueryButton
    hasQueryRan={true}
    query={[MockFunction]}
  />
</div>
`;
